src/base/iconprovider.cpp

```diff
@@ -58,11 +59,11 @@ IconProvider *IconProvider::instance()
 Path IconProvider::getIconPath(const QString &iconId) const
 {
     // there are a few icons not available in svg
-    const Path pathSvg {":/icons/" + iconId + ".svg"};
+    const Path pathSvg {u":/icons/"_qs + iconId + u".svg"_qs};
```

Comment:
This part of code was to demonstrate the literal operator works as intended.
I should (revert this) find another place for demonstration.

Reply:
> This part of code was to demonstrate the literal operator works as intended.

But we should use it in a more optimal way, especially if we're talking about optimization. We shouldn't create QString instance where it isn't required. Usually these are the same cases in which we used to use string literals or QLatin1String.
>UPDATE: done.

...with some another "not so good" examples (according to above).

Reply:
>Usually these are the same cases in which we used to use string literals or QLatin1String.

The new literal operator is slightly faster than `QLatin1String`, I don't see why we shouldn't replace it.

>with some another "not so good" examples (according to above).

Can you point out which exact change is sub-optimal?

Reply:
> > Usually these are the same cases in which we used to use string literals or QLatin1String.
> 
> The new literal operator is slightly faster than `QLatin1String` in Qt5 and a lot faster in Qt6, I don't see why we shouldn't replace it

In the cases "we used to use string literals or `QLatin1String`" before we can just use `char16_t` string literals now to avoid unnecessary creation of `QString` objects where it isn't required.
>

> Besides, `u""` literals and `u""_qs` isn't very different in binary in Qt6, I wouldn't be surprised they are stored identical in binary.

There are `char16_t` arrays stored in binary in both cases but `u""_qs` creates `QString` on top of such array.


> > We shouldn't create QString instance where it isn't required.
> 
> Right, but in the demonstration all string instances are `QString`.

No, you misunderstood it.
E.g.:
`const int index = finalPathStr.lastIndexOf(u"_new"_qs, -1, Qt::CaseInsensitive);`. We don't really need to instantiate QString from `u"_new"`.
`const Path newPath = deserialize(m_name + u"_new"_qs, res);`. We don't need to instantiate QString from `u"_new"`.
In all such cases we can just use `char16_t` string literals so the method overloads which use `QStringView` (that is more lightweight than `QString`) will be called.

Reply:
>In all such cases we can just use char16_t string literals so the method overloads which use QStringView (that is more lightweight than QString) will be called.

Where does it happen? in Qt lib or you meant we should provide `QStringView` overloads ourselves?

Reply:
>Where does it happen? in Qt lib?

Yes.
E.g.: https://doc.qt.io/qt-6/qstring.html#indexOf-3

Reply:
OK, you are right, it seems only `m_data {TransactionalSettings(u"qBittorrent"_qs).read()}` and `const TransactionalSettings settings(u"qBittorrent"_qs);` are the only valid use cases.

Reply:
Updated.